internal/file/file.go

```diff
@@ -23,33 +22,27 @@ func New[T any]() *File[T] {
 func (f *File[T]) Write(data *[]T) error {
 	bytes, err := json.Marshal(data)
 	if err != nil {
-		return fmt.Errorf("marshalling file: %w", err)
+		return fmt.Errorf("marshal file: %w", err)
 	}
 
 	err = os.WriteFile(filename, bytes, 0644)
 	if err != nil {
-		return fmt.Errorf("writing file: %w", err)
+		return fmt.Errorf("write file: %w", err)
 	}
 
 	return nil
 }
 
 // Read - reads tasks from json file
 func (f *File[T]) Read() ([]T, error) {
-	file, err := os.Open(filename)
-	if err != nil {
-		if errors.Is(err, os.ErrNotExist) {
-			return nil, nil
-		}
+	bytes, err := os.ReadFile(filename)
 
-		return nil, fmt.Errorf("opening file: %w", err)
+	if errors.Is(err, os.ErrNotExist) {
+		return nil, nil
 	}
 
```

Comment:
I usually stick all if err checks together without empty lines, so it kinda same process goes as a whole. unless there is maybe a big handler that logically outstands.